mmv1/third_party/terraform/functions/region_from_zone.go

```diff
@@ -41,11 +43,23 @@ func (f RegionFromZoneFunction) Run(ctx context.Context, req function.RunRequest
 		return
 	}
 
-	// Validate input - can't be an empty string, every zone will have 2nd to last element in string be a `-`
-	if len(arg0) != 0 && arg0[len(arg0)-2] == '-' {
-		resp.Diagnostics.Append(resp.Result.Set(ctx, arg0[:len(arg0)-2])...)
-	} else {
+	if arg0 == "" {
+		resp.Diagnostics.AddArgumentError(
+			0,
+			noMatchesErrorSummary,
+			"The input string is empty.",
+		)
 		return
 	}
 
+	if arg0[len(arg0)-2] != '-' {
+		resp.Diagnostics.AddArgumentError(
+			0,
+			noMatchesErrorSummary,
+			"The input string is invalid.",
```

Comment:
Could you please make this feedback more explicit? If we alert users that the input doesn't 'look' like a zone it can help them navigate the problem more easily. For example "The input string is invalid, please make sure you are providing the name of a zone that ends with "-[a-z]"."

Reply:
Another thing to consider is to interpolate the input string into the summary of the error. It could be that the user doesn't know what string they're passing in! It could be that a value that's being returned from an API, or something read from [remote state](https://developer.hashicorp.com/terraform/language/state/remote-state-data) that the user doesn't control themselves. In that case the feedback could just be "The input string %s is not a valid region name"